frontend/lib/src/hooks/useWidgetManagerElementState.tsx

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useCallback, useEffect, useState } from "react"
+
+import { WidgetStateManager } from "@streamlit/lib/src/WidgetStateManager"
+import {
+  isNullOrUndefined,
+  notNullOrUndefined,
+} from "@streamlit/lib/src/util/utils"
+
+const useWidgetManagerElementState = <T,>({
```

Comment:
nice addition 🥳 

nit: maybe add a small docstring comment that explains what the hook can be used for.